docs/contents/pdf_model.puml

```diff
@@ -0,0 +1,240 @@
+.. uml::
+
+    @startuml
+
+        skinparam object {
+            AttributeFontSize 11
+        }
+
+        ' increase distance between nodes
+        skinparam nodesep 50
+
+        ' default arrow color and arrow font color
+        !define COL_ARROW_DEFAULT darkblue
+        skinparam arrowcolor COL_ARROW_DEFAULT
+        skinparam arrowfontcolor COL_ARROW_DEFAULT
+        skinparam shadowing false
+
+        skinparam package<<Layout>> {
+            borderColor Transparent
+            backgroundColor Transparent
+            fontColor Transparent
+            stereotypeFontColor Transparent
+        }
+        ' color of arrows showing inheritance
+        !define COL_DERIVE red
+
+        
+        class Root {
+        --
+        **content**: container for elements of type
+        ""Chapter/Table/Figure/Paragraph""
+        --
+        **library/API**: All ""Root"" elements are exposed as given in this diagram.
+
+        **CLI/YAML/JSON**: ""Root.pages"" contain a list of ""Page"" instances without
+        ""Page.contents"" attribute. Instead, all content goes to ""Root.contents""
+        and the nested ""Chapter.contents"".
+        }
+
+        class Element {
+            + [str] id
+            + [str] type
+            --
+            **id**: unique in given scope (root or chapter)
+            **type**: parent class name
+
+            **b_root**: set, if the element
+            is directly located under root
+            **b_chapter**: set, if the element
+            is located under a chapter
+        }
+
+        class Coord {
```

Comment:
The implementation differs from the UML currently (`TextBase` missing, `Word`/`HorizontalBox`/`HorizontalBox` not inheriting from `Coord`). Without providing a target solution, I'd say the following 3 requirements should be met:

1. The UML model should 1:1 correspond to instances of class `ApiObjects` when inspecting with a debugger. There should be no missing attributes/properties and additional attributes/properties are ideally hidden/private.
2. All UML model classes should be properly documented in docs/contents/api.rst.
3. Users of libpdf can expect proper typing of all public classes and methods, that is IDEs can inspect returned types. In practice that means attributes/properties should either be class members or, in case of instance members, be created in `__init__`. Also they should provide type information.